File: deleteNode.js
// Time Complexity : O(1)
// Space Complexity : O(1)
// Did this code successfully run on GeeksForGeeks : Yes
// Any problem you faced while coding this : No

//Function to delete a node without any reference to head pointer.
class deleteNode {
    
    deleteNode(del)
    {
        //your code here
        let temp = del.next;
        del.data = temp.data;
        del.next = temp.next;
    }
}

File: DeleteNode.cs
// Time Complexity : O(1)
// Space Complexity : O(1)
// Did this code successfully run on geeks : yes
// Any problem you faced while coding this : no

class Solution
    {
        //Complete this function
        public void deleteNode(Node del)
        {
            //Your code here
            Node remove = del.next;
            del.data = remove.data;
            del.next = del.next.next;
        }

    }
